### ISSUE TITLE
[Help] Can't get Notebook setup to work

### ISSUE THREAD

**Issue by erezak** [benlubas/molten-nvim#222]
I've carefully gone through the setup instructions in neovim.
In a previous try (a couple of months ago) I managed to make it work, but I've recreated my config since.

Now, for the life of me, I can't run anything.
It either says that the coderunner isn't initiliazed, or - if I try to run quarto.activate in the ftplugin for markdown, I get this:

```
Error detected while processing BufReadCmd Autocommands for "*.ipynb":                                                                                                              
Error executing lua callback: ...ocal/share/nvim/lazy/jupytext.nvim/lua/jupytext/init.lua:150: BufReadCmd Autocommands for "*.ipynb"..FileType Autocommands for "*"..function <SNR>1
_LoadFTPlugin[19]..script /Users/ek844y/.config/nvim/after/ftplugin/markdown.lua: Vim(runtime):E5113: Error while calling lua chunk: ...al/share/nvim/lazy/otter.nvim/lua/otter/lsp/
handlers.lua:26: attempt to index local 'ms' (a nil value)                                                                                                                          
stack traceback:                                                                                                                                                                    
        ...al/share/nvim/lazy/otter.nvim/lua/otter/lsp/handlers.lua:26: in main chunk                                                                                               
        [C]: in function 'require'                                                                                                                                                  
        ....local/share/nvim/lazy/otter.nvim/lua/otter/lsp/init.lua:2: in main chunk                                                                                                
        [C]: in function 'require'                                                                                                                                                  
        ...44y/.local/share/nvim/lazy/otter.nvim/lua/otter/init.lua:7: in main chunk                                                                                                
        [C]: in function 'require'                                                                                                                                                  
        ...y/.local/share/nvim/lazy/quarto-nvim/lua/quarto/init.lua:108: in function 'activate'                                                                                     
        /Users/ek844y/.config/nvim/after/ftplugin/markdown.lua:3: in main chunk                                                                                                     
        [C]: in function 'nvim_command'                                                                                                                                             
        ...ocal/share/nvim/lazy/jupytext.nvim/lua/jupytext/init.lua:150: in function 'read_from_ipynb'                                                                              
        ...ocal/share/nvim/lazy/jupytext.nvim/lua/jupytext/init.lua:176: in function <...ocal/share/nvim/lazy/jupytext.nvim/lua/jupytext/init.lua:175>                              
stack traceback:                                                                                                                                                                    
        [C]: in function 'nvim_command'                                                                                                                                             
        ...ocal/share/nvim/lazy/jupytext.nvim/lua/jupytext/init.lua:150: in function 'read_from_ipynb'                                                                              
        ...ocal/share/nvim/lazy/jupytext.nvim/lua/jupytext/init.lua:176: in function <...ocal/share/nvim/lazy/jupytext.nvim/lua/jupytext/init.lua:175>                              
Press ENTER or type command to continue
```

Here is my lazy config file. For the life of me, I can't understand the issue.

Thanks!

```
return {
  {
    "benlubas/molten-nvim",
    version = "^1.0.0", -- use version <2.0.0 to avoid breaking changes
    dependencies = { "3rd/image.nvim" },
    build = ":UpdateRemotePlugins",
    init = function()
      -- this is an example, not a default. Please see the readme for more configuration options
      -- I find auto open annoying, keep in mind setting this option will require setting
      -- a keybind for `:noautocmd MoltenEnterOutput` to open the output again
      vim.g.molten_auto_open_output = false

      -- this guide will be using image.nvim
      -- Don't forget to setup and install the plugin if you want to view image outputs
      vim.g.molten_image_provider = "image.nvim"

      -- optional, I like wrapping. works for virt text and the output window
      vim.g.molten_wrap_output = true

      -- Output as virtual text. Allows outputs to always be shown, works with images, but can
      -- be buggy with longer images
      vim.g.molten_virt_text_output = true

      -- this will make it so the output shows up below the \`\`\` cell delimiter
      vim.g.molten_virt_lines_off_by_1 = true
      vim.g.molten_output_win_max_height = 20
    end,
    config = function()
      vim.keymap.set(
        "n",
        "<localleader>e",
        ":MoltenEvaluateOperator<CR>",
        { desc = "evaluate operator", silent = true }
      )
      vim.keymap.set(
        "n",
        "<localleader>os",
        ":noautocmd MoltenEnterOutput<CR>",
        { desc = "open output window", silent = true }
      )
      vim.keymap.set(
        "n",
        "<localleader>rr",
        ":MoltenReevaluateCell<CR>",
        { desc = "re-eval cell", silent = true }
      )
      vim.keymap.set(
        "v",
        "<localleader>r",
        ":<C-u>MoltenEvaluateVisual<CR>gv",
        { desc = "execute visual selection", silent = true }
      )
      vim.keymap.set(
        "n",
        "<localleader>oh",
        ":MoltenHideOutput<CR>",
        { desc = "close output window", silent = true }
      )
      vim.keymap.set("n", "<localleader>md", ":MoltenDelete<CR>", { desc = "delete Molten cell", silent = true })

      -- if you work with html outputs:
      vim.keymap.set(
        "n",
        "<localleader>mx",
        ":MoltenOpenInBrowser<CR>",
        { desc = "open output in browser", silent = true }
      )
    end,
  },
  -- plugins/quarto.lua
  {
    "quarto-dev/quarto-nvim",
    dependencies = {
      "jmbuhr/otter.nvim",
      "nvim-treesitter/nvim-treesitter",
    },
    ft = { "quarto", "markdown" },
    config = function()
      local quarto = require("quarto")
      quarto.setup({
        lspFeatures = {
          -- NOTE: put whatever languages you want here:
          languages = { "r", "python", "rust" },
          chunks = "all",
          diagnostics = {
            enabled = true,
            triggers = { "BufWritePost" },
          },
          completion = {
            enabled = true,
          },
        },
        keymap = {
          -- NOTE: setup your own keymaps:
          hover = "H",
          definition = "gd",
          rename = "<leader>rn",
          references = "gr",
          format = "<leader>gf",
        },
        codeRunner = {
          enabled = true,
          default_method = "molten",
        },
      })
      local runner = require("quarto.runner")
      vim.keymap.set("n", "<localleader>rc", runner.run_cell, { desc = "run cell", silent = true })
      vim.keymap.set("n", "<localleader>ra", runner.run_above, { desc = "run cell and above", silent = true })
      vim.keymap.set("n", "<localleader>rA", runner.run_all, { desc = "run all cells", silent = true })
      vim.keymap.set("n", "<localleader>rl", runner.run_line, { desc = "run line", silent = true })
      vim.keymap.set("v", "<localleader>r", runner.run_range, { desc = "run visual range", silent = true })
      vim.keymap.set("n", "<localleader>RA", function()
        runner.run_all(true)
      end, { desc = "run all cells of all languages", silent = true })
    end,
  },
  {
    "GCBallesteros/jupytext.nvim",
    config = function()
      require("jupytext").setup({
        style = "markdown",
        output_extension = "md",
        force_ft = "markdown",
      })
    end,
  },
}
```

**Comment by benlubas** [comment]
What version of neovim are you on? It looks like your version doesn't have `vim.lsp.protocol.Methods`.

Confirm that's the issue with `:= vim.lsp.protocol.Methods`. It should be a table, but for you it will probably be nil.

If so, you can fix it by updating to version 0.10.1.

Depending on your system and package manager that might not be trivial. [Bob](https://github.com/MordechaiHadad/bob) makes managing nvim versions really easy.

**Comment by erezak** [comment]
Darn. That was it.
Thank you.

Let me take this opportunity to tell you again that it's a work of art. Thanks for making this plugin
